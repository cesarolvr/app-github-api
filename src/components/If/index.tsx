import React from "react";

interface IfProps extends React.PropsWithChildren {
  condition: boolean;
}

const If = ({ children, condition }: IfProps): JSX.Element | null => {
  return <>{condition ? children : null}</>;
};

export default If;
