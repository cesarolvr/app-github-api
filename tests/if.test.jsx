import renderer from "react-test-renderer";

// Component
import If from "../src/components/If/index";

it("render true component", () => {
  const component = renderer.create(
    <If
      condition={true}
      renderIf={<>True condition</>}
      renderElse={<>Falsy condition</>}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render falsy component", () => {
  const component = renderer.create(
    <If
      condition={false}
      renderIf={<>True condition</>}
      renderElse={<>Falsy condition</>}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
