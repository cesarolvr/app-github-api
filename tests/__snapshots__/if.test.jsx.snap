// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render falsy component 1`] = `"Falsy condition"`;

exports[`render true component 1`] = `"True condition"`;
